File: 3Star BegLvl/MAXCOMP.py
def maxCompensation():
    n = int(input())
    v = []
    for i in range(n):
        s, e, c = map(int, input().split())
        v.append([s, e, c])
    
    dp = [0] * 49
    for i in range(1, 49):
        max_val = dp[i - 1]
        for j in range(n):
            if v[j][1] == i:
                max_val = max(max_val, v[j][2] + dp[v[j][0]])
        dp[i] = max_val
    print(dp[48])

def main():
    t = int(input())
    for _ in range(t):
        maxCompensation()

if __name__ == "__main__":
    main()

File: 3Star BegLvl/MAXCOMP.cpp
#include <bits/stdc++.h>
using namespace std;

void maxCompensation() {
    int n;
    cin >> n;
    int s, e, c;
    vector<vector<int>> v;
    for (int i=0; i<n; ++i) {
        cin >> s >> e >> c;
        v.push_back({s,e,c});
    }
    
    vector<int> dp(49,0);
    for (int i=1; i<=48; ++i) {
        int max_val=dp[i-1];
        for (int j=0; j<n; ++j) {
            if (v[j][1]==i) {
                max_val = max(max_val,v[j][2]+dp[v[j][0]]);
            }
        }
        dp[i] = max_val;
    }
    cout << dp[48] << endl;
}

int main() {
	// your code goes here
	int t;
	cin >> t;
	while (t--) {
	    maxCompensation();
	}
	return 0;
}
